### PR TITLE
Improvement/replace router package

### PR DESCRIPTION
**Pull request by danyecls**

### CODE REVIEW

File: agent/pkg/tunnel/tunnel.go [hunk 1]
```diff
@@ -5,47 +5,47 @@ import (
...
 		},
 	}
-	t.router.HandleFunc("/ssh/http", func(w http.ResponseWriter, r *http.Request) {
-		t.HTTPHandler(w, r)
+	e.GET("/ssh/http", func(e echo.Context) error {
+		return t.HTTPHandler(e)
 	})
-	t.router.HandleFunc("/ssh/{id}", func(w http.ResponseWriter, r *http.Request) {
-		t.ConnHandler(w, r)
+	e.GET("/ssh/{id}", func(e echo.Context) error {
```
Comment (gustavosbarreto) [inline agent/pkg/tunnel/tunnel.go:44]: Path parameters in Echo has a different annotation: https://echo.labstack.com/guide/request/#path-parameters

File: agent/main.go [hunk 2]
```diff
@@ -142,91 +143,78 @@ func NewAgentServer() *Agent { // nolint:gocyclo
...
+		if err != nil {
+			return c.String(http.StatusInternalServerError, "failed to hijack connection")
 		}
 
-		serv.Sessions[vars["id"]] = conn
-		serv.HandleConn(conn)
+		vars := c.ParamValues()
+		//preciso fechar esse httpConn?
+		httpConn := c.Request().Context().Value("http-conn").(net.Conn)
+		serv.Sessions[vars[0]] = httpConn
```
Comment (gustavosbarreto) [inline agent/main.go:160]: `c.Param("id")` instead of `c.ParamValues()[0]`

File: agent/main.go [hunk 3]
```diff
@@ -142,91 +143,78 @@ func NewAgentServer() *Agent { // nolint:gocyclo
...
-			return
+		conn, _, err := hj.Hijack()
+		if err != nil {
+			return c.String(http.StatusInternalServerError, "failed to hijack connection")
 		}
 
-		serv.Sessions[vars["id"]] = conn
-		serv.HandleConn(conn)
+		vars := c.ParamValues()
+		//preciso fechar esse httpConn?
```
Comment (gustavosbarreto) [inline agent/main.go:158]: ?

File: agent/main.go [hunk 4]
```diff
@@ -142,91 +143,78 @@ func NewAgentServer() *Agent { // nolint:gocyclo
...
+
+		return nil
 	}
-	tun.CloseHandler = func(w http.ResponseWriter, r *http.Request) {
-		vars := mux.Vars(r)
-		serv.CloseSession(vars["id"])
+
+	tun.CloseHandler = func(c echo.Context) error {
+		vars := c.ParamNames()
+		serv.CloseSession(c.Param(vars[0]))
```
Comment (gustavosbarreto) [inline agent/main.go:215]: `c.Param()`

File: agent/pkg/tunnel/tunnel.go (hunk 1 above)
Comment (danyecls) [reply agent/pkg/tunnel/tunnel.go]: Corrected for correct echo syntax

File: agent/main.go (hunk 2 above)
Comment (danyecls) [reply agent/main.go]: Fixed instance of c.Param("id)

File: agent/main.go (hunk 3 above)
Comment (danyecls) [reply agent/main.go]: Removed dead comment from code

File: agent/main.go (hunk 4 above)
Comment (danyecls) [reply agent/main.go]: Fixed instance of c.Param("id)

File: ssh/main.go [hunk 5]
```diff
@@ -37,105 +36,92 @@ func main() {
 	tunnel := sshTunnel.NewTunnel("/ssh/connection", "/ssh/revdial")
 
 	router := tunnel.GetRouter()
-	router.HandleFunc("/sessions/{uid}/close", func(response http.ResponseWriter, request *http.Request) {
-		exit := func(response http.ResponseWriter, status int, err error) {
-			log.WithError(err).WithFields(log.Fields{
-				"status": status,
-			}).Error("failed to close the session")
+	router.POST("/sessions/:uid/close", func(c echo.Context) error {
```
Comment (henrybarreto) [inline ssh/main.go:39]: I and @gustavosbarreto have seen it must be `.Any` instead of `.POST`.

Comment (danyecls) [reply ssh/main.go]: correct to Any

**Pull request merged**